hivemq-edge/src/frontend/src/components/rjsf/MqttTransformation/utils/json-schema.utils.ts

```diff
@@ -1,61 +1,91 @@
-      if (defName) {
-        list.push({ title: title || key, type: 'object', description, path: [] })
-        if (definitions?.[defName]) {
-          // TODO[NVL] This is incorrect! It should use the same path as below
-          const { properties: subs } = definitions[defName] as JSONSchema7
-          if (subs)
-            for (const [key, value] of Object.entries(subs)) {
-              const { type: type2, title: title2 } = value as JSONSchema7
-              list.push({ title: title2 || key, type: type2, path: [title || ''] })
-            }
-        }
-      }
+  // check if it's a ref to a definition
+  const { $ref: ref, title } = property as JSONSchema7
+  if (ref) {
+    // We only support the local definitions, i.e. with pattern "#/definitions/{property}"
+    const defPath = ref.split('/')
+    const defName = defPath.pop()
+    if (ref.startsWith('#') && defName && definitions?.[defName]) {
+      tempProperty = definitions?.[defName]
     } else {
-      list.push({ title: title || key, type, description, path: [] })
-      if (type === 'object') {
-        const { properties } = property as JSONSchema7
-        if (properties) {
-          for (const [key, value] of Object.entries(properties)) {
-            const { type: type2, title: title2 } = value as JSONSchema7
-            list.push({ title: title2 || key, type: type2, path: [title || ''] })
-          }
-        }
-      }
-      if (type === 'array') {
-        const { items } = property as JSONSchema7
-        if (items) {
-          const { properties: subs, type: arrayItemType } = items as FlatJSONSchema7
-          if (subs)
-            for (const [key, value] of Object.entries(subs)) {
-              const { type: type2, title: title2 } = value as JSONSchema7
-              list.push({ title: title2 || key, type: type2, path: [title || ''] })
-            }
-          else {
-            list.push({ title: ARRAY_ITEM_INDEX || key, type: arrayItemType, path: [title || ''] })
-          }
-        }
+      // This is an error, ref defined but definition not found.
+      // No error in JSONSchema so using null type and description
+      return [{ key: key, path: path, type: 'null', description: `error: definition ${defName} doesn't exist` }]
+    }
+  }
+
+  // Add the property
+  const { type, description, title: titleRef, examples } = tempProperty as JSONSchema7
+  const mainProps: FlatJSONSchema7 = { key, path, type, description, title: titleRef || title || key, examples }
+
+  const subProperties: FlatJSONSchema7[] = []
+  if (type === 'object') {
+    // Check recursively for object's properties
+    const { properties } = tempProperty as JSONSchema7
+    if (properties) {
+      for (const [subKey, subProp] of Object.entries(properties)) {
+        subProperties.push(...getProperty(subKey, [...path, key], subProp, definitions))
       }
     }
+  } else if (type === 'array') {
+    // Check recursively for item's properties
```

Comment:
is there code missing here? If not, should this `else if` be removed?

Reply:
code missing